exchanges/bitfinex/bitfinex_websocket.go

```diff
@@ -1521,15 +1521,57 @@ func (b *Bitfinex) WsUpdateOrderbook(p currency.Pair, assetType asset.Item, book
 				err)
 		}
 
-		err = validateCRC32(ob, checkme.Token)
-		if err != nil {
+		if err = validateCRC32(ob, checkme.Token); err != nil {
+			log.Errorf(log.WebsocketMgr, "%s websocket orderbook update error, will resubscribe orderbook: %v", b.Name, err)
+			if suberr := b.resubOrderbook(p, assetType); suberr != nil {
+				log.Errorf(log.ExchangeSys, "%s error resubscribing orderbook: %v", b.Name, suberr)
+			}
 			return err
 		}
 	}
 
 	return b.Websocket.Orderbook.Update(&orderbookUpdate)
 }
 
+// resubOrderbook resubscribes the orderbook after a consistency error, probably a failed checksum,
+// which forces a fresh snapshot. If we don't do this the orderbook will keep erroring and drifting.
+func (b *Bitfinex) resubOrderbook(p currency.Pair, assetType asset.Item) error {
+	if err := b.Websocket.Orderbook.FlushOrderbook(p, asset.Spot); err != nil {
+		return err
+	}
+
+	c, err := b.chanForSub(wsBook, assetType, p)
+	if err != nil {
+		return err
+	}
+	return b.Websocket.ResubscribeToChannel(c)
+}
+
+// chanForSub pulls together the original request channel with Params
+// Along with the
+func (b *Bitfinex) chanForSub(cName string, assetType asset.Item, pair currency.Pair) (*stream.ChannelSubscription, error) {
```

Comment:
I think this function can get some test coverage

Reply:
I'm in the process of working out how to remove it, actually ...

Reply:
I'm a few tasks away from making chanForSub redundant.
But I don't want to fold that work together with this PR.
Do you reckon we can skip the further testing of chanForSub in-lieu of that?

Reply:
I'm happy to wait for the fix. Otherwise, an easy way to make this testable is to pass in the subs as a param so you don't need to do any `Websocket` sub setup for test case scenarios

Reply:
I see you've mastered the dark art of baiting me into writing baits.
Promoted to Lvl 12 Warlock 🧙 

Test added.